docs.markdown

```diff
@@ -116,6 +116,140 @@ Response is again a JSON object, similar to the original one:
 
 You can use the refresh token multiple times until the refresh token expires itself and you need to go through the process again.
 
+### Multi-Factor Authentication
+
+To have access to Kontist API endpoints that require strong customer authentication, you need to pass Multi-Factor Authentication (MFA).
+
+We provide a simplified push notification MFA flow for users who have installed the Kontist Application and paired their device in it.
+Alternatively, if you have setup [Device Binding](/docs/advanced-authentication#device-binding) in your application, you can also implement your own MFA flow on top of it using our API's endpoints.
+
+#### Creating a challenge
+
+To initiate the MFA procedure, you will need to create an MFA Challenge:
+
+```shell
+curl "https://api.kontist.com/api/user/mfa/challenges" \
+  -H "Authorization: Bearer ey..." \
+  -X POST
+```
+
+
+> The above command returns JSON structured like this:
+```json
+{
+  "id": "5f7c36e2-e0bf-4755-8376-ac6d0711192e",
+  "status": "PENDING",
+  "expiresAt": "2019-12-02T16:25:15.933+00:00"
+}
+```
+
+##### HTTP Request
+
+`POST https://api.kontist.com/api/user/mfa/challenges`
+
+##### Response
+
+| Field      | Description                                                           |
+| ---------- | --------------------------------------------------------------------- |
+| id         | ID of the challenge                                                   |
+| status     | Status of the challenge. When created, it will be "PENDING"           |
+| expiresAt  | Time at which the challenge will expire                               |
+
+
+#### Verifying a challenge
+
+The next step to pass MFA is to verify the challenge that was just created.
+
+If the user has the Kontist Application on his device, he will receive a push notification prompting him to "Confirm login". After logging into the application and confirming, the challenge will be verified (its status will be updated to `VERIFIED`).
+If you are using the push notification flow that we provide, you can skip to the [next step](#polling-for-challenge-verification)
+
+If you are setting up your own MFA flow, the challenge can also be verified by accessing this endpoint:
+
+```shell
+curl "https://api.kontist.com/api/user/mfa/challenges/5f7c36e2-e0bf-4755-8376-ac6d0711192e" \
+  -H "Authorization: Bearer ey..." \
+  -H "Content-Type: application/json" \
+  -X PATCH \
+  -d '{
+        "status": "VERIFIED"
+      }'
+```
+
+**Note:** in order for a challenge to be verified, this endpoint needs to be accessed using a *confirmed* token that you can obtain with [Device Binding](/docs/advanced-authentication#device-binding)
+
+> The above command returns `204 No Content` in case of success.
+
```

Comment:
Setting the status to "VERIFIED" looks like a very special use case (only used by our mobile application and will be hard to do for others as only the mobile application will get the push notifications). Or do I miss sth. here?
I suggest to move that into the "advanced-authentication" page.

Reply:
Yeah it is a very specific case and most likely not usable in any practical way I agree. I will definitely move it 👍

The case that I had in mind is that a dev could have an application where he implemented device binding (so the users can have confirmed access tokens there), and another one where he doesn't have device binding (e.g. a web app). 
In this case he could replicate the MFA flow that we have for our own web app, for instance by sending sms or his own push notification before / after hitting the endpoint to create the challenge, without the need to use the kontist app at all. Quite the edge case though yeah 😄

Reply:
Have extracted that part here: 5457347b3131affea3da5e5283a215e2b51814bb

How does it look?

I'm unsure if we should reference it with a link on the main doc or not 🤔

Reply:
Looks good and the use case you described makes sense. Would not add a special link in the main doc.

Reply:
Yeah sounds good, I removed the link: 8b5f113e14241afeee6a04083a46a1309ac7ea31

Then I think we are good to merge it 🙂